src/pumipic_adjacency.tpp

```diff
@@ -466,7 +467,8 @@ namespace pumipic {
                    o::Write<o::Real>& inter_points,
                    int looplimit,
                    bool debug,
-                   Func& func) {
+                   Func& func,
+                   o::Real &tol) {
```

Comment:
Can `tol` be made an optional argument to avoid the 'checking for negative' logic?  Something like this:
https://godbolt.org/z/Poa5haq6d